magic.js

```diff
@@ -1147,6 +1148,201 @@ function uid(sec, cb) {
 }
 
 
+/*****************
+ *    Streams    *
+ *****************/
+/* The libsodium library expects to decrypt the stream in the same chunks it
+ * encrypted it. I've set the stream chunk size in 4KB, so that libsodium
+ * encrypts/decrypts stream chunks of 4KB each time except from the last one. The
+ * 4KB chunk is ok for most scenarios (e.g. encrypting/decrypting files) but not
+ * ideal for real time stream manipulation. Adding support for that would require
+ * to store the length of each received chunk in the encrypted stream. This means
+ * additional implementation work and thus is omitted for now.
+ */
+const STREAM_CHUNK_SIZE = exports.STREAM_CHUNK_SIZE = 4096;
+
+/* The first byte of the encrypted stream will always indicate the version of the
+ * EncryptStream. For now it's set to 1. This will allow us to modify the data we
+ * store in the EncryptStream in the future and mark the change with a new
+ * version.
+ */
+const STREAM_VERSION = 1;
+const KEY_SIZE = 32
```

Comment:
We should probably keep the semicolon rule consistent unless there's a javascript convention I'm not aware of.